### PR TITLE
Improved handling for Identities on no wires

### LINKED ISSUE #5532: [BUG] Cannot run qnode with sum that contains `I()` (Identity with empty wires)
Getting an error for a `Sum` containing `I()`, despite https://github.com/PennyLaneAI/pennylane/pull/5414

```python
dev = qml.device("default.qubit")

h = qml.dot([0.5, 0.5], [qml.I(), qml.X(0)])
assert isinstance(h, qml.ops.Sum)

@qml.qnode(dev)
def qnode():
    return qml.expval(h)

qnode()
```

```pycon
File ~/Xanadu/pennylane/pennylane/measurements/measurements.py:174, in MeasurementProcess.__init__(self, obs, wires, eigvals, id)
    172 if wires is not None:
    173     if len(wires) == 0:
--> 174         raise ValueError("Cannot set an empty list of wires.")
    175     if obs is not None:
    176         raise ValueError("Cannot set the wires if an observable is provided.")

ValueError: Cannot set an empty list of wires.
```

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

While we have added the ability to have `qml.I()` on no wires to better reflect the wire-independent nature of the identity and create operations again from a simplified form, we don't necessarily have extensive support and coverage for this new style of observable yet.

**Description of the Change:**

This PR extends support for `qml.I()` for `devices.qubit` measurement pathways `measure` and `measure_with_samples`. It also allows us to calculate the eigenvalues for composite ops that contain an operator on no wires.  And finally, it adds support for observables with no wires to `qml.pauli.group_observables`.

Throughout, I assume an observable with no wires commutes with everything, as `qml.I` does in fact commute with everything.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-61509] Fixes #5532

### CODE REVIEW

**Comment by Qottmann** [comment]
Awesome! Seems to fix my issues with qnodes that compute expectation values with CompositeOps that contain I() instances 👌 

```
@qml.qnode(qml.device("default.qubit"))
def qnode():
    return qml.expval(I() + Z(0))

qnode()
```

**Review by Qottmann — COMMENTED** [review]
Awesome, thanks so much @albi3ro 

Just missing one integration test with a qnode perhaps? Otherwise happy to approve!

File: pennylane/pauli/grouping/group_observables.py [hunk 1]
```diff
@@ -177,6 +177,37 @@ def colour_pauli_graph(self):
         return self.grouped_paulis
 
 
+def _calculate_partitioned_coefficients(observables, partitioned_paulis, coefficients):
```
Comment (albi3ro) [inline pennylane/pauli/grouping/group_observables.py:180]: Just extracting some code to get pylint to stop complaining.

File: pennylane/measurements/expval.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ def circuit(x):
     Returns:
         ExpectationMP: measurement process instance
     """
+    if isinstance(op, qml.Identity) and len(op.wires) == 0:
+        # temporary solution to merge https://github.com/PennyLaneAI/pennylane/pull/5106
+        raise NotImplementedError(
+            "Expectation values of qml.Identity() without wires are currently not allowed."
+        )
+
```
Comment (Qottmann) [inline pennylane/measurements/expval.py:64]: So one can measure `qml.expval(1. * qml.I())` but not `qml.expval(qml.I())`? Couldnt we just do

```suggestion
    if isinstance(op, qml.Identity) and len(op.wires) == 0:
        return ExpectationMP(obs=1.*op)

```
perhaps?

**Review by Qottmann — COMMENTED** [review]
One minor point about straight `I()` not being allowed to be measured

File: pennylane/measurements/expval.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/measurements/expval.py:64]: So this was just me accidentally removing this logic and then re-adding it in a different place.  

While we now mostly support `ExpectationMP(qml.I())`, I think we should still have the error in `qml.expval` as a guard against something that might be more problematic.  We could even consider expanding the error to cover `qml.expval(2*qml.I())`.


Mostly works now, but I think we should probably do more validation and checking before making it available to a generic user.

Comment (albi3ro) [reply pennylane/measurements/expval.py]: plugins. and everything else. As this PR shows, a lot of places weren't equipped to handle it.

**Review by Qottmann — APPROVED** [review]
Great PR, thanks so much for the fix @albi3ro ! 🥂 

My main concern of Hamiltonians/Sums containing I() being able to be handled is solved with this 👍 

Would be good to also be able to directly measure I() as well, but for the moment this will do!

File: pennylane/pauli/grouping/group_observables.py (hunk 1 above)
Comment (astralcai) [reply pennylane/pauli/grouping/group_observables.py:180]: I did the same thing in https://github.com/PennyLaneAI/pennylane/pull/5559, you might need to merge from master and resolve some merge conflicts.

**Review by astralcai — APPROVED** [review]

**Pull request merged**